TestGolden{Add,Init}Cmd time-dependent thus failing reproducibility test

TestGoldenAddCmd and TestGoldenInitCmd compares newly generated files containing current year with pre-generated "golden" files set in 2017.  This fails "Reproducible builds" as explained in https://reproducible-builds.org/.

From https://tests.reproducible-builds.org/debian/logs/unstable/arm64/golang-github-spf13-cobra_0.0~git20170724.0.90fc11b-3.build2.log.gz (transient URL):

(or https://web.archive.org/web/20170729172939/https://tests.reproducible-builds.org/debian/logs/unstable/arm64/golang-github-spf13-cobra_0.0~git20170724.0.90fc11b-3.build2.log.gz for a permanent URL):

```
I: Current time: Sat Sep  1 09:01:29 +14 2018

...

dpkg-source: info: extracting golang-github-spf13-cobra in 2nd
dpkg-source: info: unpacking golang-github-spf13-cobra_0.0~git20170724.0.90fc11b.orig.tar.xz
dpkg-source: info: unpacking golang-github-spf13-cobra_0.0~git20170724.0.90fc11b-3.debian.tar.xz
dpkg-source: info: applying 001-fix-TestGlobalNormFuncPropagation-on-gccgo.patch

...

   dh_auto_test -O--buildsystem=golang
	go test -v -p 8 github.com/spf13/cobra github.com/spf13/cobra/cobra github.com/spf13/cobra/cobra/cmd github.com/spf13/cobra/doc

...

=== RUN   TestGoldenAddCmd
--- FAIL: TestGoldenAddCmd (0.01s)
	add_test.go:69: "/build/golang-github-spf13-cobra-0.0~git20170724.0.90fc11b/2nd/obj-aarch64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go" and "testdata/root.go.golden" are not equal!
		
		$ diff /build/golang-github-spf13-cobra-0.0~git20170724.0.90fc11b/2nd/obj-aarch64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go testdata/root.go.golden
		1c1
		< // Copyright © 2018 NAME HERE <EMAIL ADDRESS>
		---
		> // Copyright © 2017 NAME HERE <EMAIL ADDRESS>
		
		exit status 1
=== RUN   TestValidateCmdName
--- PASS: TestValidateCmdName (0.00s)
=== RUN   TestGoldenInitCmd
--- FAIL: TestGoldenInitCmd (0.01s)
	init_test.go:67: "/build/golang-github-spf13-cobra-0.0~git20170724.0.90fc11b/2nd/obj-aarch64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go" and "testdata/root.go.golden" are not equal!
		
		$ diff /build/golang-github-spf13-cobra-0.0~git20170724.0.90fc11b/2nd/obj-aarch64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go testdata/root.go.golden
		1c1
		< // Copyright © 2018 NAME HERE <EMAIL ADDRESS>
		---
		> // Copyright © 2017 NAME HERE <EMAIL ADDRESS>
		
		exit status 1
=== RUN   TestFindExistingPackage
--- PASS: TestFindExistingPackage (0.00s)
FAIL
FAIL	github.com/spf13/cobra/cobra/cmd	0.052s
```

I wonder how to keep these tests while satisfying reproducibility?
Would some kind of fake time.Now work?

--- a/cobra/cmd/add_test.go
+++ b/cobra/cmd/add_test.go
@@ -6,6 +6,8 @@ import (
 	"os"
 	"path/filepath"
 	"testing"
+
+	"github.com/spf13/viper"
 )
 
 // TestGoldenAddCmd initializes the project "github.com/spf13/testproject"
@@ -20,6 +22,9 @@ func TestGoldenAddCmd(t *testing.T) {
 	// Initialize the project at first.
 	initializeProject(project)
 	defer os.RemoveAll(project.AbsPath())
+	defer viper.Set("year", nil)
+
+	viper.Set("year", 2017) // For reproducible builds
 
 	// Then add the "test" command.
 	cmdName := "test"

--- a/cobra/cmd/init_test.go
+++ b/cobra/cmd/init_test.go
@@ -6,6 +6,8 @@ import (
 	"os"
 	"path/filepath"
 	"testing"
+
+	"github.com/spf13/viper"
 )
 
 // TestGoldenInitCmd initializes the project "github.com/spf13/testproject"
@@ -16,6 +18,9 @@ func TestGoldenInitCmd(t *testing.T) {
 	projectName := "github.com/spf13/testproject"
 	project := NewProject(projectName)
 	defer os.RemoveAll(project.AbsPath())
+	defer viper.Set("year", nil)
+
+	viper.Set("year", 2017) // For reproducible builds
 
 	os.Args = []string{"cobra", "init", projectName}
 	if err := rootCmd.Execute(); err != nil {

--- a/cobra/cmd/licenses.go
+++ b/cobra/cmd/licenses.go
@@ -77,7 +77,11 @@ func getLicense() License {
 
 func copyrightLine() string {
 	author := viper.GetString("author")
-	year := time.Now().Format("2006")
+
+	year := viper.GetString("year") // For reproducible builds
+	if year == "" {
+		year = time.Now().Format("2006")
+	}
 
 	return "Copyright © " + year + " " + author
 }

